api/v1alpha1/workflow_webhook.go

```diff
@@ -162,15 +171,15 @@ func (w *Workflow) ValidateUpdate(old runtime.Object) error {
 	}
 
 	if newState < oldState {
-		return fmt.Errorf("state cannot progress backwards")
+		return field.Invalid(specPath.Child("DesiredState"), w.Spec.DesiredState, "DesiredState cannot progress backwards")
 	}
 
 	if newState > oldState+1 {
-		return fmt.Errorf("states cannot be skipped")
+		return field.Invalid(specPath.Child("DesiredState"), w.Spec.DesiredState, "states cannot be skipped")
 	}
 
 	if !oldWorkflow.Status.Ready {
-		return fmt.Errorf("current desired state not yet achieved")
+		return field.Invalid(statusPath.Child("State"), oldWorkflow.Status.State, "current desired state not yet achieved")
```

Comment:
You're checking on ready but then reporting on state; is that intentional?

Reply:
Yes, as the person debugging this, I'd like to know what the state is that is not ready.